index.js

```diff
@@ -212,7 +214,17 @@ function tsLookup({dependency, filename, tsConfig}) {
     options.module = ts.ModuleKind.AMD;
   }
 
-  const host = ts.createCompilerHost({});
+  var host = ts.createCompilerHost({});
+
+  // Override host methods, to support finding files in provided fs.
+  const isDirectory = dirPath => fileSystem.lstatSync(dirPath).isDirectory();
+  const getDirectories = dirPath => fileSystem.readdirSync(dirPath).map(name => path.join(dirPath, name)).filter(isDirectory);
+  const pathExists = filePath => fileSystem.existsSync(filePath);
+
+  host.getDirectories = getDirectories;
```

Comment:
Should we only do these overrides if a custom filesystem is provided? And do you think there is a path forward for putting this functionality into the TS compiler proper?

Reply:
As per above, we could implement a custom compilerHost, with these methods. This would be cleaner than monkey patching. We could then use the default compilerHost as before when no custom fileSystem, or our own compilerHost when there is a custom fileSystem. It's a bit more effort, and will add code to maintain in this code base for the custom compiler host, and I am not sure the gains would be much. I'll follow your preference.

Reply:
I was just hoping for a method that minimizes the surface are of what needs to be maintained from my end. If these overrides are all that's necessary, then it's fine for now. Thanks for the reasoning.